test/mongodb.test.js

```diff
@@ -1032,6 +1032,16 @@ describe('mongodb connector', function() {
   });
 
   describe('updateAll', function() {
+    it('should preserve the id', async function() {
```

Comment:
IIUC, this test is verifying that when the `id` is provided as string (e.g. when coming from the HTTP request body), then the connector is correctly coercing it to `ObjectID` value. Please capture this important information in the test name.

You may want to add another test to verify what happens when properties are modified directly in the model instance.

```ts
const user = await User.create({name: 'Al', age: 31, email: 'al@strongloop'});
const userId = user.id;
user.age = 100;

 await User.update(user);
// ...
```

Reply:
> ... then the connector is correctly coercing it to ObjectID value

Nah, this test is just ensuring the `id` property is not deleted.

> You may want to add another test to verify what happens when properties are modified directly in the model instance.

Isn't it already happening on the model instance? 😃

Reply:
> > You may want to add another test to verify what happens when properties are modified directly in the model instance.
>
> Isn't it already happening on the model instance?

No, you are making the change on the object returned by `user.toObject()`. That's a new object instance with prototype set to `Object`, it's not the model instance with prototype set to `User`.

Reply:
Updated.